samples/bluetooth/broadcast_audio_source: FLASH overflow when built for nrf5340dk_nrf5340_cpuapp

**Describe the bug**

`samples/bluetooth/broadcast_audio_source` overflows FLASH region when built for `nrf5340dk_nrf5340_cpuapp`

**To Reproduce**

`west build -b nrf5340dk_nrf5340_cpuapp samples/bluetooth/broadcast_audio_source -T sample.bluetooth.broadcast_audio_source -p`

Using legacy uart shim for hci_ipc fixes the problem:
`west build -b nrf5340dk_nrf5340_cpuapp samples/bluetooth/broadcast_audio_source -T sample.bluetooth.broadcast_audio_source -p -- -Dhci_ipc_CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=y`

FLASH delta seems to be of about 1544 bytes.

**Expected behavior**
Sample builds successfully

**Impact**
Sample broken, impacts CI for other PRs (see https://github.com/zephyrproject-rtos/zephyr/actions/runs/7574708555/job/20629782144?pr=67673).

**Logs and console output**

```
...
Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -DPICOLIBC_INTEGER_PRINTF_SCANF  --specs=picolibc.specs  -lc  -lgcc && cd /home/gmarull/ws/nordic/zephyrproject/zephyr/build/hci_ipc/zephyr && /usr/bin/cmake -E true
/home/gmarull/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `datas' will not fit in region `FLASH'
/home/gmarull/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: region `FLASH' overflowed by 300 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
[13/16] Completed 'broadcast_audio_source'
FAILED: hci_ipc-prefix/src/hci_ipc-stamp/hci_ipc-build /home/gmarull/ws/nordic/zephyrproject/zephyr/build/hci_ipc-prefix/src/hci_ipc-stamp/hci_ipc-build 
cd /home/gmarull/ws/nordic/zephyrproject/zephyr/build/hci_ipc && /usr/bin/cmake --build .
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.1
 - Commit SHA or Version used b6e48d04f62d3aeb2afa40a2fab38baae3ea5709

**Additional context**
N/A

--- a/drivers/serial/Kconfig.nrfx
+++ b/drivers/serial/Kconfig.nrfx
@@ -31,9 +31,8 @@ config UART_NRFX_UARTE
 config UART_NRFX_UARTE_LEGACY_SHIM
 	bool "Legacy UARTE shim"
 	depends on UART_NRFX_UARTE
-	# New shim takes more ROM. Until it is fixed use legacy shim in memory
-	# constraint case.
-	default y if MCUBOOT
+	# New shim takes more ROM. Until it is fixed use legacy shim.
+	default y
 
 config UART_ASYNC_TX_CACHE_SIZE
 	int "TX cache buffer size"

--- a/tests/drivers/uart/uart_async_api/testcase.yaml
+++ b/tests/drivers/uart/uart_async_api/testcase.yaml
@@ -13,7 +13,6 @@ tests:
     harness_config:
       fixture: gpio_loopback
     depends_on: gpio
-    tags: bsim_skip_CI
   drivers.uart.wide:
     filter: CONFIG_SERIAL_SUPPORT_ASYNC and not CONFIG_UART_MCUX_LPUART
     harness: ztest
@@ -26,7 +25,7 @@ tests:
     platform_allow: nucleo_h743zi
     integration_platforms:
       - nucleo_h743zi
-  drivers.uart.async_api.nrf_uarte_legacy:
+  drivers.uart.async_api.nrf_uarte_new:
     platform_allow: nrf52840dk_nrf52840 nrf52_bsim
     filter: CONFIG_SERIAL_SUPPORT_ASYNC
     harness: ztest
@@ -34,7 +33,8 @@ tests:
       fixture: gpio_loopback
     depends_on: gpio
     extra_configs:
-      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=y
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
+    tags: bsim_skip_CI
   drivers.uart.async_api.nrf_uart:
     filter: CONFIG_SERIAL_SUPPORT_ASYNC
     harness: ztest

--- a/tests/drivers/uart/uart_mix_fifo_poll/testcase.yaml
+++ b/tests/drivers/uart/uart_mix_fifo_poll/testcase.yaml
@@ -19,6 +19,7 @@ tests:
       - CONFIG_UART_INTERRUPT_DRIVEN=n
       - CONFIG_UART_ASYNC_API=n
       - CONFIG_UART_0_ENHANCED_POLL_OUT=n
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.uart_mix_poll_fifo:
@@ -26,6 +27,7 @@ tests:
       - CONFIG_UART_INTERRUPT_DRIVEN=y
       - CONFIG_UART_0_INTERRUPT_DRIVEN=y
       - CONFIG_UART_0_ENHANCED_POLL_OUT=n
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.uart_mix_poll_async_api:
@@ -34,6 +36,7 @@ tests:
       - CONFIG_UART_0_INTERRUPT_DRIVEN=n
       - CONFIG_UART_0_ASYNC=y
       - CONFIG_UART_0_ENHANCED_POLL_OUT=n
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.uart_mix_poll_async_api_const:
@@ -44,6 +47,7 @@ tests:
       - CONFIG_UART_0_ASYNC=y
       - CONFIG_UART_0_ENHANCED_POLL_OUT=n
       - CONFIG_UART_0_TX_CACHE_SIZE=2
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI # We skip a few tests to save CI time, as they give little extra coverage
 
   drivers.uart.uart_mix_poll_with_ppi:
@@ -51,6 +55,7 @@ tests:
       - CONFIG_UART_INTERRUPT_DRIVEN=n
       - CONFIG_UART_ASYNC_API=n
       - CONFIG_UART_0_ENHANCED_POLL_OUT=y
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.uart_mix_poll_fifo_with_ppi:
@@ -58,6 +63,7 @@ tests:
       - CONFIG_UART_INTERRUPT_DRIVEN=y
       - CONFIG_UART_0_INTERRUPT_DRIVEN=y
       - CONFIG_UART_0_ENHANCED_POLL_OUT=y
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.uart_mix_poll_async_api_with_ppi:
@@ -66,6 +72,7 @@ tests:
       - CONFIG_UART_0_INTERRUPT_DRIVEN=n
       - CONFIG_UART_0_ASYNC=y
       - CONFIG_UART_0_ENHANCED_POLL_OUT=y
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
     tags: bsim_skip_CI
 
   drivers.uart.legacy.uart_mix_poll:

